# HG changeset patch
# User Felix Janda <felix.janda@posteo.de>
# Date 1423173146 -3600
#      Thu Feb 05 22:52:26 2015 +0100
# Node ID c084550fcab646d911bbf4f05747f228759161e4
# Parent  2d344ef8da33717224d5a08570dc834ca2254f9f
profiler: avoid getcontext and gettid

diff -r 2d344ef8da33 -r c084550fcab6 tools/profiler/GeckoTaskTracer.cpp
--- a/tools/profiler/GeckoTaskTracer.cpp	Sun Feb 01 19:32:36 2015 +0100
+++ b/tools/profiler/GeckoTaskTracer.cpp	Thu Feb 05 22:52:26 2015 +0100
@@ -18,8 +18,8 @@
 
 #include <stdarg.h>
 
-#if defined(__GLIBC__)
-// glibc doesn't implement gettid(2).
+#if defined(OS_LINUX) && !defined(ANDROID)
+// only bionic wraps gettid(2).
 #include <sys/syscall.h>
 static pid_t gettid()
 {
diff -r 2d344ef8da33 -r c084550fcab6 tools/profiler/platform-linux.cc
--- a/tools/profiler/platform-linux.cc	Sun Feb 01 19:32:36 2015 +0100
+++ b/tools/profiler/platform-linux.cc	Thu Feb 05 22:52:26 2015 +0100
@@ -92,8 +92,8 @@
 
 #define SIGNAL_SAVE_PROFILE SIGUSR2
 
-#if defined(__GLIBC__)
-// glibc doesn't implement gettid(2).
+#if !defined(ANDROID)
+// only android wraps gettid(2).
 #include <sys/syscall.h>
 pid_t gettid()
 {
